### PR TITLE
Graceyi/hotfix/housekeeping

### PR DESCRIPTION
**Pull request by yunqingyi**

### CODE REVIEW

File: ex2_hal/athena/equipment_handler/source/housekeeping_athena.c [hunk 1]
```diff
@@ -1,141 +1,181 @@
...
-    for (i = 0; i < 2; i++) {
-        athena_hk->temparray[i] = (long)csp_ntoh32((uint32_t)athena_hk->temparray[i]);
-    }
-    athena_hk->boot_cnt = csp_ntoh16(athena_hk->boot_cnt);
-    athena_hk->OBC_uptime = csp_ntoh16(athena_hk->OBC_uptime);
-    athena_hk->cmds_received = csp_ntoh16(athena_hk->cmds_received);
-    athena_hk->pckts_uncovered_by_FEC = csp_ntoh16(athena_hk->pckts_uncovered_by_FEC);
-    return 0;
-}
+/*
```
Comment (Taylorr82) [inline ex2_hal/athena/equipment_handler/source/housekeeping_athena.c:1]: This whole file got changed to windows line endings, probably some other files too. It needs to be changed back to unix line endings

File: ex2_hal/athena/equipment_handler/source/housekeeping_athena.c [hunk 2]
```diff
@@ -1,141 +1,181 @@
...
+ * @return
+ * 		integer OBC_uptime to store Athena uptime
+ *      Seconds = OBC_uptime*10
+ *      Max = 655350 seconds (7.6 days)
+ */
+uint16_t Athena_get_OBC_uptime() {
+
+    TickType_t OBC_ticks = xTaskGetTickCount(); // 1 tick/ms
+    uint32_t OBC_uptime_32 = OBC_ticks / 1000;
+    if (OBC_uptime_32 > 655350) {
```
Comment (Taylorr82) [inline ex2_hal/athena/equipment_handler/source/housekeeping_athena.c:55]: This doesn't make sense. Why is there a cap on uptime?

File: ex2_hal/athena/equipment_handler/source/housekeeping_athena.c [hunk 3]
```diff
@@ -1,141 +1,181 @@
...
+ * @brief
+ *      Getter to Athena solar panel supply current
+ * @return
+ * 		  integer solar_panel_supply_curr to store solar panel supply current
+ */
+uint16_t Athena_get_solar_supply_curr() {
+    // insert getter function for solar panel supply current;
+    uint8_t addr = 0b1000101;
+    uint16_t retval;
+    int get_status = ina209_get_current(addr, &retval);
```
Comment (Taylorr82) [inline ex2_hal/athena/equipment_handler/source/housekeeping_athena.c:76]: missing error check

File: ex2_hal/athena/equipment_handler/source/housekeeping_athena.c [hunk 4]
```diff
@@ -1,141 +1,181 @@
...
+    iErr = red_statvfs("VOL0:", &volstat);
+    athena_hk->vol0_usage_percent =
+        (uint8_t)((float)(volstat.f_bfree) * 100.0 /
+                  ((float)(volstat.f_blocks))); // assuming block size == sector size = 512B
+    if (iErr == -1) {
+        exit(red_errno);
+    }
+
+#ifdef IS_ATHENA_V2
+    iErr = red_statvfs("VOL1:", &volstat);
```
Comment (Taylorr82) [inline ex2_hal/athena/equipment_handler/source/housekeeping_athena.c:140]: Does VOL1: work here or does it need the trailing /?

**Review by Taylorr82 — CHANGES_REQUESTED** [review]

File: ex2_hal/athena/equipment_handler/source/housekeeping_athena.c (hunk 1 above)
Comment (yunqingyi) [reply ex2_hal/athena/equipment_handler/source/housekeeping_athena.c:1]: :( not again

File: ex2_hal/athena/equipment_handler/source/housekeeping_athena.c (hunk 2 above)
Comment (yunqingyi) [reply ex2_hal/athena/equipment_handler/source/housekeeping_athena.c:55]: the cap is to convert it from uint32 to uint16

Comment (Taylorr82) [reply ex2_hal/athena/equipment_handler/source/housekeeping_athena.c:55]: The uptime size should be uint32. It doesn't make sense to delete information

Comment (yunqingyi) [reply ex2_hal/athena/equipment_handler/source/housekeeping_athena.c:55]: this is to convert uint32 to uint16

Comment (Taylorr82) [reply ex2_hal/athena/equipment_handler/source/housekeeping_athena.c:55]: I know, I'm saying it shouldn't, the uptime should be uint32. It gets converted to seconds I see, which makes sense. The largest 32 bit unsigned integer when divided by 1000 creates an integer that can be represented in 3 bytes, or 24 bits. We don't want to mess with 3 byte integers, so 32 bits should be used for the uptime

Comment (yunqingyi) [reply ex2_hal/athena/equipment_handler/source/housekeeping_athena.c]: changed from uint16 to uint32

File: ex2_hal/athena/equipment_handler/source/housekeeping_athena.c (hunk 3 above)
Comment (yunqingyi) [reply ex2_hal/athena/equipment_handler/source/housekeeping_athena.c]: how do you error check errors from driver code? Do you want to check the current range?

File: ex2_hal/athena/equipment_handler/source/housekeeping_athena.c (hunk 4 above)
Comment (yunqingyi) [reply ex2_hal/athena/equipment_handler/source/housekeeping_athena.c]: I just used the path prefix in gaRedVolConf, which didn't have the trailing /, someone would need to test it on athena

File: ex2_hal/athena/equipment_handler/source/ina209.c [hunk 5]
```diff
@@ -1,215 +1,215 @@
...
-//         // Read current from register
-//         ina209_get_current(SOLAR_INA209_ADDR, &current_reg);
-////        ina209_get_shunt_voltage(SOLAR_INA209_ADDR, &shunt_voltage);
-//        printf("Current Register:\t0x%.4x -> %d \r\n", current_reg, current_reg);
-//        vTaskDelay(500);
-//    }
-//    printf("Current Test Complete\r\n");
-//    return rtn;
-//}
+/*
```
Comment (Taylorr82) [inline ex2_hal/athena/equipment_handler/source/ina209.c:1]: This file is still CRLF

Comment (yunqingyi) [reply ex2_hal/athena/equipment_handler/source/ina209.c:1]: should be ok now

Comment (Taylorr82) [reply ex2_hal/athena/equipment_handler/source/ina209.c:1]: The commit has no content

File: ex2_hal/athena/equipment_handler/source/housekeeping_athena.c (hunk 2 above)
Comment (yunqingyi) [reply ex2_hal/athena/equipment_handler/source/housekeeping_athena.c]: the comments before my second comment didn't show up lol, so now I look like a bot, i'm not a bot for the record.

File: ex2_hal/athena/equipment_handler/source/housekeeping_athena.c [hunk 6]
```diff
@@ -33,33 +32,26 @@ const uint8_t software_version = 3;
...
 /**
  * @brief
  *      Getter to Athena uptime
  * @return
  * 		integer OBC_uptime to store Athena uptime
  *      Seconds = OBC_uptime*10
- *      Max = 655350 seconds (7.6 days)
  */
-uint16_t Athena_get_OBC_uptime() {
+uint32_t Athena_get_OBC_uptime() {
```
Comment (joshdellaz) [inline ex2_hal/athena/equipment_handler/source/housekeeping_athena.c:49]:
```suggestion
uint32_t get_obc_uptime_seconds() {
```

Comment (joshdellaz) [reply ex2_hal/athena/equipment_handler/source/housekeeping_athena.c:49]: "get" at the front is slightly preferable

Comment (joshdellaz) [reply ex2_hal/athena/equipment_handler/source/housekeeping_athena.c:49]: maybe just preference tho

File: ex2_hal/athena/equipment_handler/source/housekeeping_athena.c [hunk 7]
```diff
@@ -70,9 +62,12 @@ uint16_t Athena_get_OBC_uptime() {
  * @return
  * 		  integer solar_panel_supply_curr to store solar panel supply current
  */
-uint8_t Athena_get_solar_supply_curr() {
+uint16_t Athena_get_solar_supply_curr() {
```
Comment (joshdellaz) [inline ex2_hal/athena/equipment_handler/source/housekeeping_athena.c:65]: append _raw or _mA to the function name to specify the return value format

File: ex2_hal/athena/equipment_handler/source/housekeeping_athena.c [hunk 8]
```diff
@@ -90,31 +85,69 @@ uint8_t Athena_get_solar_supply_curr() {
...
-    // placeholder for software version
-    athena_hk->OBC_software_ver = software_version;
-
-    if (temporary != 0)
-        return_code = temporary;
+    if (temp_status != 0)
+        return_code = temp_status;
 
     // Get solar panel supply current
     athena_hk->solar_panel_supply_curr = Athena_get_solar_supply_curr();
```
Comment (joshdellaz) [inline ex2_hal/athena/equipment_handler/source/housekeeping_athena.c:118]: redundant with line 112

File: ex2_hal/athena/equipment_handler/source/housekeeping_athena.c [hunk 9]
```diff
@@ -90,31 +85,69 @@ uint8_t Athena_get_solar_supply_curr() {
...
+    iErr = red_statvfs("VOL1:", &volstat);
+    athena_hk->vol1_usage_percent =
+        (uint8_t)((float)(volstat.f_bfree) * 100.0 /
+                  ((float)(volstat.f_blocks))); // assuming block size == sector size = 512B
+    if (iErr == -1) {
+        exit(red_errno);
+    }
+
+#else  // IS_ATHENA_V2
+    athena_hk->vol1_usage_percent = 0; // stub if no card
```
Comment (joshdellaz) [inline ex2_hal/athena/equipment_handler/source/housekeeping_athena.c:142]:
```suggestion
    athena_hk->vol1_usage_percent = 255; // stub if no card
```

Comment (joshdellaz) [reply ex2_hal/athena/equipment_handler/source/housekeeping_athena.c:142]: This makes it clearer that there is no SD card as opposed to an SD card that is 0% full. Do the same for line 146

File: ex2_hal/athena/equipment_handler/source/tempsense_athena.c [hunk 10]
```diff
@@ -19,27 +19,27 @@
 
 #include "tempsense_athena.h"
 #include "tmp421.h"
```
Comment (joshdellaz) [inline ex2_hal/athena/equipment_handler/source/tempsense_athena.c:21]:
```suggestion
```

File: ex2_services/Services/source/housekeeping/housekeeping_service.c [hunk 11]
```diff
@@ -227,15 +228,13 @@ Result mock_everyone(All_systems_housekeeping *all_hk_data) {
...
-        all_hk_data->Athena_hk.temparray[i] = tempLong;
-    }
+    all_hk_data->Athena_hk.MCU_core_temp = tempLong;
+    all_hk_data->Athena_hk.converter_temp = tempLong;
     all_hk_data->Athena_hk.boot_cnt = tempu16;
     all_hk_data->Athena_hk.OBC_mode = tempu8;
     all_hk_data->Athena_hk.OBC_uptime = tempu16;
     all_hk_data->Athena_hk.solar_panel_supply_curr = tempu8;
-    all_hk_data->Athena_hk.OBC_software_ver = tempu8;
+    memset(all_hk_data->Athena_hk.OBC_software_ver, 0, 8 * sizeof(char));
```
Comment (joshdellaz) [inline ex2_services/Services/source/housekeeping/housekeeping_service.c:237]: Suggest having a variable or macro for 8*sizeof(char) titled something like "version_id_size" for clarity

File: ex2_hal/athena/equipment_handler/source/housekeeping_athena.c (hunk 6 above)
Comment (Taylorr82) [reply ex2_hal/athena/equipment_handler/source/housekeeping_athena.c:49]: Get at the front is better

**Comment by Taylorr82** [comment]
Replaced by https://github.com/AlbertaSat/ex2_obc_software/pull/293

**Pull request closed without merging**